src/requests/publishAssetAsync.luau

```diff
@@ -25,28 +30,83 @@ local function publishAssetAsync(
 	local existingAssetId = if existingAsset then existingAsset.assetId else nil
 	local assetDetails = if existingAssetId then getAssetDetailsAsync(existingAssetId, apiKey) else nil
 
-	local variables = runLuauTask(publishAssetTask, {
-		CREATOR_ID = environment.creatorId,
-		CREATOR_TYPE = environment.creatorType,
-		ASSET_TYPE = asset.type or "Package",
+	local creatorIdKey = if environment.creatorType == "User" then "userId" else "groupId"
+
+	local assetId
+
+	if assetDetails and assetDetails.assetId then
+		logging.info("publishing new version...")
+		local formData = createFormData({
+			request = {
+				value = {
+					assetId = assetDetails.assetId,
+				},
+				contentType = "application/json",
+			},
+			fileContent = {
+				filename = asset.model,
+				value = fs.readFile(asset.model),
+				contentType = "model/x-rbxm",
+			},
+		})
+
+		local res = fetch(`https://apis.roblox.com/assets/v1/assets/{assetDetails.assetId}`, {
+			method = "PATCH",
+			headers = {
+				["x-api-key"] = apiKey,
+				["Content-Type"] = `multipart/form-data; boundary={formData.boundary}`,
+			},
+			body = formData.body,
+		})
+
+		local body = serde.decode("json", res.body)
 
-		-- Only provide the assetId from the lockfile if there's a matching
-		-- asset on the Creator Store. We need this check in case the lockfile
-		-- is edited manually to a nonexistent id
-		ASSET_ID = if assetDetails and assetDetails.assetId then existingAssetId else nil :: any,
-	}, {
-		universeId = environment.universeId,
-		placeId = environment.placeId,
-		modelPath = asset.model,
-		apiKey = apiKey,
-	})
+		local operationRes = waitForAssetOperationAsync(body.operationId, apiKey)
 
-	local assetId = if variables then variables.ASSET_ID else nil
+		if operationRes and operationRes.response then
+			assetId = operationRes.response.assetId
+		end
+	else
+		logging.info("creating asset for the first time...")
+		local formData = createFormData({
+			request = {
+				value = {
+					assetType = "Model", -- TODO: Allow for publishing Plugins
+					displayName = asset.name,
+					creationContext = {
+						creator = {
+							[creatorIdKey] = environment.creatorId,
+						},
+					},
+				},
+				contentType = "application/json",
+			},
+			fileContent = {
+				filename = asset.model,
+				value = fs.readFile(asset.model),
+				contentType = "model/x-rbxm",
+			},
+		})
+
+		local res = fetch("https://apis.roblox.com/assets/v1/assets", {
```

Comment:
we have this url in two spots - i don't think it's any better stored in a constant at this point, but if we get more endpoint URLs in here might be worth slapping then in a constant file or up top. i dont' think this tips the scale yet this is just me musing about web-platform patterns